### PR TITLE
4WIF: Change 3: 4wif implementation from betaflight

### PR DESCRIPTION
**Pull request by mlyle**
Plus adaptations to the code for dRonin.  Depends upon #1779 and #1780.

### CODE REVIEW

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

(0xnone)

**Comment by mlyle** [comment]
jenkins, build artifacts please

File: flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c [hunk 1]
```diff
@@ -178,7 +185,7 @@ struct msp_bridge {
 #if defined(PIOS_MSP_STACK_SIZE)
 #define STACK_SIZE_BYTES PIOS_MSP_STACK_SIZE
 #else
-#define STACK_SIZE_BYTES 768
+#define STACK_SIZE_BYTES 1500
```
Comment (tracernz) [inline flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:188]: Not great on F1/3 :(

**Comment by tracernz** [comment]
Can we please move the BF code to a "3rdparty" subdir or flight/Libraries? Not a good look if that's the first code somebody sees in our tree and thinks it's ours. :fearful:

**Comment by mlyle** [comment]
Rebased, did a hacky shim thing.  Need to test / verify this still works.

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
Test 1 -- readout of setup from Lux + Linux + BlHeliConfigurator OK.

**Comment by mlyle** [comment]
Test 2 -- OG Brain + Linux + BlHeliConfigurator-- failed.  Can not open serial port in any form on Linux+OG Brain.

**Comment by mlyle** [comment]
Test 3 -- Seppuku + Linux + BlHeliConfigurator -- pass.

**Comment by mlyle** [comment]
Test2 Redux: programmed 4 ESCs with OG Brain + Linux + BlHeliConfigurator from 14.4 to 14.9 and reconfigured; though one of the flashes failed the first time around.

**Comment by mlyle** [comment]
Test 4-- Lux + Windows + BlHeliSuite pass-- wrote config to 4 ESCs changing beacon strength.

Required undoing a change made while fumbling around for BlHeli32 support-- where newer MSP protocol number causes MSP_MOTOR_CONFIG instead of MSP_MOTORS.

**Comment by mlyle** [comment]
Test 5-- OG Brain + Windows + BlHeliSutie pass-- wrote config to 4 ESCs changing beep strength.

At this point, I recommend merge-- possible there's some edges/targets that don't work but everything tested looks good.

**Review by tracernz — APPROVED** [review]
Looks okay on our side. Didn't look hard at BF code because it made me sad.

File: flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c (hunk 1 above)
Comment (mlyle) [reply flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:188]: Yah, F1 doesn't matter.  Probably can be lowered with more analysis but works OK on F3.

Comment (tracernz) [reply flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:188]: Does PipX not relay MSP?

Comment (mlyle) [reply flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:188]: No UAVOMSPBridge on Pipx, no.  What would that be used for?  ground side OSD?

Comment (tracernz) [reply flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:188]: Yeah, that's about it. Let's not worry about it since it doesn't exist. :D

**Pull request merged**